packages/p2p/src/utils/orders.js

```diff
@@ -1,177 +1,273 @@
+
+        // Keep this here, has_timer_expired should take priority over statuses below.
+        if (this.is_expired_order || this.has_timer_expired) {
+            return localize('Expired');
+        }
+
+        if (this.is_pending_order) {
+            const wait_for_payment = localize('Wait for payment');
+            const pay_now = localize('Pay now');
+
+            if (this.is_my_ad) {
+                return this.is_buy_order ? wait_for_payment : pay_now;
+            }
+
+            return this.is_buy_order ? pay_now : wait_for_payment;
+        }
+
+        if (this.is_buyer_confirmed_order) {
+            const confirm_payment = localize('Confirm payment');
+            const wait_for_release = localize('Wait for release');
+
+            if (this.is_my_ad) {
+                return this.is_buy_order ? confirm_payment : wait_for_release;
+            }
+
+            return this.is_buy_order ? wait_for_release : confirm_payment;
+        }
+
+        return localize('Unknown');
+    }
+
+    get labels() {
+        if (this.is_buy_order) {
+            if (this.is_my_ad) {
+                return {
+                    other_party_role: localize('Buyer'),
+                    left_send_or_receive: localize('Receive'),
+                    right_send_or_receive: localize('Send'),
+                    payment_details: localize('Your payment details'),
+                    contact_details: localize('Your contact details'),
+                    instructions: localize('Your instructions'),
+                    result_string: localize('You sold {{offered_amount}} {{offered_currency}}', {
+                        offered_amount: this.order_details.amount_display,
+                        offered_currency: this.order_details.account_currency,
+                    }),
+                };
+            }
+
+            return {
+                other_party_role: localize('Seller'),
+                left_send_or_receive: localize('Send'),
+                right_send_or_receive: localize('Receive'),
+                payment_details: localize("Seller's payment details"),
+                contact_details: localize("Seller's contact details"),
+                instructions: localize("Seller's instructions"),
+                result_string: localize("You've received {{offered_amount}} {{offered_currency}}", {
+                    offered_amount: this.order_details.amount_display,
+                    offered_currency: this.order_details.account_currency,
+                }),
+            };
         }
-    } else if (is_buyer_cancelled_order) {
-        status_string = localize('Cancelled');
-    } else if (is_buyer_confirmed_order) {
-        if (is_my_ad) {
-            status_string = is_buy_order ? localize('Confirm payment') : localize('Wait for release');
-        } else {
-            status_string = is_buy_order ? localize('Wait for release') : localize('Confirm payment');
+
+        if (this.is_my_ad /* && !this.is_buy_order */) {
```

Comment:
Is the comment intentional?

Reply:
It is. It provides a bit of extra context for the next person.

Reply:
Looks like you missed it! 😅  
Can you add it above or outside the condition?

Reply:
The comment was put there intentionally, but I'll remove it since it's not doing what it's supposed to do. 👍